scripts/util/doc-builder.js

```diff
@@ -0,0 +1,236 @@
+        }
+
+        promise = promise.then(() => {
+          console.info('info:', 'Getting commit for', name);
+          if(isMasterBranch) {
+            return repo.getBranchCommit(name);
+          }
+
+          return repo.getCommit(target);
+        }).then(commit => {
+          console.info('info:', 'Getting tree associated with commit on', name);
+
+          return commit.getTree();
+        }).then(tree => {
+          var checkoutOptions = {
+            checkoutStrategy: this.options.checkoutStrategy,
+            paths: this.options.srcDirName,
+            targetDirectory: path.join(repo.workdir(), this.options.apiDirPath, name)
+          };
+
+          paths.push(checkoutOptions.targetDirectory);
+          console.info('info:', 'Checking out the tree on', name);
+
+          return NodeGit.Checkout.tree(repo, tree, checkoutOptions);
+        });
+      });
+
+      return promise.then(() => {
+        console.info('info:', 'Done checking out source code');
+        return paths;
+      });
+    });
+  });
+}
+
+DocBuilder.prototype.generateAPIDocs = function (paths) {
+  console.info('\ninfo:', 'Generating API docs');
+  return yuidoc.generate({paths: paths, themeDir: this.options.themeDir}).then(() => {
+    console.info('info:', 'Done generating API docs');
+  });
+
+}
+
+DocBuilder.prototype.commitAPIDocs = function () {
+  var apiTempDir = path.join('.', this.options.versionsDirName);
+
+  console.info('\ninfo:', 'Commiting API Docs');
+  if (fs.existsSync(apiTempDir)) {
+    fsExtra.removeSync(apiTempDir);
+  }
+
+  console.info('info:', 'Copying', this.options.apiDirPath, 'to root directory');
+  fsExtra.copySync(this.options.apiDirPath, apiTempDir);
+
+  return NodeGit.Repository.open('.').then(repo => {
+
+    console.info('info:', 'Getting repo\'s current staging area');
+    return repo.index().then(index => {
+      console.info('info:', 'Getting', this.options.docsBranchName, '\'s latest commit');
+      return repo.getBranchCommit(this.options.docsBranchName).then(docsBranchCommit => {
+        console.info('info:', 'Getting the commit\'s tree');
+        return docsBranchCommit.getTree().then(tree => {
+          console.info('info:', 'Loading the tree into the staging area');
+          return index.readTree(tree);
+        }).then(() => {
+          console.info('info:', 'Staging \'', this.options.versionsDirName, '\' to \'', this.options.docsBranchName, '\'');
+          return index.addAll(`${this.options.versionsDirName}`, NodeGit.Index.ADD_OPTION.ADD_FORCE);
+        }).then(() => {
+          return repo.getReference(this.options.docsBranchName);
+        }).then(reference => {
+          var docsBranchResolvedName = reference.name();
+          console.info('info:', 'Blew up', this.options.docsBranchName, 'into', docsBranchResolvedName);
+          console.info('info:', 'Getting the staging area tree id');
+          return index.writeTree().then(treeId => {
+            var dateObject = new Date();
```

Comment:
eep! Are you sure you want to do the date stuff yourself, rather than use a sane lib like momentJS?